pythran/pythonic/builtins/bin.hpp

```diff
@@ -6,29 +6,47 @@
 #include "pythonic/types/str.hpp"
 #include "pythonic/utils/functor.hpp"
 
-#include <sstream>
+#include <cmath>
+#include <limits>
 
 PYTHONIC_NS_BEGIN
 
 namespace builtins
 {
   template <class T>
-  types::str bin(T const &v)
+  typename std::enable_if<std::is_scalar<T>::value, types::str>::type
+  bin(T const &v)
   {
-    long unsigned int i = 1L << (8 * sizeof(T) - 1);
-    while (i && !(v & i))
-      i >>= 1;
-    if (!i)
+    using UT = typename std::make_unsigned<T>::type;
+    if (v < T{0})
+      if (v == std::numeric_limits<T>::min()) {
+        // In this special case, calling -v would overflow so
```

Comment:
it would be great to have a test case for that path.

Reply:
I have added one, using `sys.maxsize,` however it causes tests to fail on windows.

During the tests, the parameter it treated as a `long`. And long range is different on windows:  it goes from  -2147483648 to 2147483647, when it goes from -9223372036854775808 to 9223372036854775807 on my linux system. Basically, to have consistent  behavior, `long long` should be used.

Basically as it is, this test case (`test_bin3`) should be platform dependent. I guess you have had this kind of question raised in the past. What is your take on it?

Reply:
I have pushed a test (`pythran.tests.test_base.test_3`) for it, but disabled for windows.